libsolidity/ast/ASTJsonConverter.cpp

```diff
@@ -323,7 +323,11 @@ bool ASTJsonConverter::visit(FunctionDefinition const& _node)
 {
 	std::vector<pair<string, Json::Value>> attributes = {
 		make_pair("name", _node.name()),
-		make_pair(m_legacy ? "constant" : "isDeclaredConst", _node.stateMutability() == StateMutability::View),
+		make_pair(
+			m_legacy ? "constant" : "isDeclaredConst",
+			_node.stateMutability() == StateMutability::View ||
+			_node.stateMutability() == StateMutability::Pure
```

Comment:
Not sure I understand. `constant` / `isDeclaredConst` is used by static analyzers and I would like to break them rather later than sooner. We have an additional field that provides the state mutability, so all the information is there.